File: adventjs/2025/11/solution.ts
function findUnsafeGifts(warehouse: string[]): number {
  const rows = warehouse.length;
  if (rows === 0) return 0;
  const cols = warehouse[0].length;
  
  let unsafeCount = 0;
  
  // Directions: up, down, left, right
  const directions: number[][] = [[-1, 0], [1, 0], [0, -1], [0, 1]];
  
  for (let r = 0; r < rows; r++) {
    for (let c = 0; c < cols; c++) {
      if (warehouse[r][c] === '*') {
        // Check if this present has any adjacent camera
        let hasCamera = false;
        
        for (const [dr, dc] of directions) {
          const newR = r + dr;
          const newC = c + dc;
          
          // Check if position is valid and contains a camera
          if (newR >= 0 && newR < rows && newC >= 0 && newC < cols) {
            if (warehouse[newR][newC] === '#') {
              hasCamera = true;
              break;
            }
          }
        }
        
        // If no camera found, this present is unsafe
        if (!hasCamera) {
          unsafeCount++;
        }
      }
    }
  }
  
  return unsafeCount;
}

File: adventjs/2025/11/solution.py
def find_unsafe_gifts(warehouse: list[str]) -> int:
  rows = len(warehouse)
  if rows == 0:
    return 0
  cols = len(warehouse[0])
  
  unsafe_count = 0
  
  # Directions: up, down, left, right
  directions = [(-1, 0), (1, 0), (0, -1), (0, 1)]
  
  for r in range(rows):
    for c in range(cols):
      if warehouse[r][c] == '*':
        # Check if this present has any adjacent camera
        has_camera = False
        
        for dr, dc in directions:
          new_r = r + dr
          new_c = c + dc
          
          # Check if position is valid and contains a camera
          if 0 <= new_r < rows and 0 <= new_c < cols:
            if warehouse[new_r][new_c] == '#':
              has_camera = True
              break
        
        # If no camera found, this present is unsafe
        if not has_camera:
          unsafe_count += 1
  
  return unsafe_count
